draft-ietf-jose-hpke-encrypt.md

```diff
@@ -253,15 +254,53 @@ Otherwise, the JWE Protected Header (and JWE Shared Unprotected Header) MUST NOT
 - JOSE Header parameter "psk_id" MAY be present.
 - JOSE Header parameter "ek" MUST be present and contain the base64url-encoded HPKE encapsulated key.
 - Recipient JWE Encrypted Key MUST be the ciphertext from HPKE Encryption.
-- The HPKE info parameter defaults to the empty string; mutually known private information MAY be used instead.
-- The HPKE AAD parameter MUST be set to the empty string.
+- The HPKE info parameter contains the encoding of the Recipient_structure, which is described in {{recipient_structure}}.
+- The HPKE AAD parameter defaults to the empty string; externally provided information MAY be used instead.
 - THE HPKE plaintext MUST be set to the CEK.
 
 The processing of "enc", "iv", "tag", "aad", and "ciphertext" is as already defined in {{RFC7516}}.
 Implementations process these parameters as defined in {{RFC7516}};
 no additional processing requirements are introduced by HPKE-based key encryption.
 
-## JSON Example
+## Recipient_structure {#recipient_structure}
+
+The `Recipient_structure` is a JSON object with the following members:
+
+- context (string): This member MUST include the constant string value "JOSE HPKE Recipient".
+
+- next_layer_alg (string): Identifies the algorithm with which the HPKE-encrypted key MUST be used. Its value MUST match the "enc" (encryption algorithm) header parameter in the next lower JOSE layer.
+
+- recipient_protected_header (object): This member contains the base64url-encoded header of the recipient structure. To serialize the header the procedure from Section 3.3 of RFC 7638 MUST be used. Unlike with RFC 7638, all members from the recpient header are included except for the "ek" member.
```

Comment:
Does it address any security threat or it is done only to align with COSE HPKE draft ?

Reply:
The recipient_protected_header incorporates information from the recipient header and can thereby include the identities of the endpoint in the key derivation function. There are attacks that can be constructed if you do not include this type of context information.

Reply:
And the approach is also aligned with COSE HPKE

Reply:
I updated the text and I hope it is better now.